controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -289,21 +290,47 @@ func (r *NodeNetworkConfigurationPolicyReconciler) initializeEnactment(policy nm
 		enactmentConditions := enactmentconditions.New(r.APIClient, enactmentKey)
 		enactmentConditions.Reset()
 	}
+
+	return &enactment, nil
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) fillInEnactmentStatus(policy nmstatev1.NodeNetworkConfigurationPolicy,
+	enactment nmstatev1beta1.NodeNetworkConfigurationEnactment,
+	enactmentConditions enactmentconditions.EnactmentConditions) (ctrl.Result, error) {
 	nns, err := r.readNNS(nodeName)
 	if err != nil {
-		return nil, err
+		return ctrl.Result{}, err
 	}
-	capturedStates, desiredStateMetaInfo, desiredState, err := nmpolicy.GenerateState(desiredStateWithDefaults, policy.Spec, nns.Status.CurrentState, enactment.Status.CapturedStates)
+
+	capturedStates, desiredStateMetaInfo, generatedDesiredState, err := nmpolicy.GenerateState(policy.Spec.DesiredState, policy.Spec, nns.Status.CurrentState, enactment.Status.CapturedStates)
 	if err != nil {
-		return nil, err
+		err = enactmentstatus.Update(r.APIClient, nmstateapi.EnactmentKey(nodeName, policy.Name), func(status *nmstateapi.NodeNetworkConfigurationEnactmentStatus) {
+			status.PolicyGeneration = policy.Generation
+		})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		enactmentConditions.NotifyGenerateFailure(err)
+		return ctrl.Result{Requeue: false}, err
 	}
 
-	return &enactment.Status.Conditions, enactmentstatus.Update(r.APIClient, enactmentKey, func(status *nmstateapi.NodeNetworkConfigurationEnactmentStatus) {
-		status.DesiredState = desiredState
+	desiredStateWithDefaults, err := nmstate.ApplyDefaultVlanFiltering(generatedDesiredState)
+	if err != nil {
+		return ctrl.Result{Requeue: false}, errors.Wrap(err, "error applying defaults to policy desiredState")
```

Comment:
This is tricky, we can't return an error here as the Requeue value is ignored. See the reque logic in controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/blob/8f43f26d53ffb08a96ffce1298df0cd761f8be0d/pkg/internal/controller/controller.go#L311

If err != nil , the Request is requeued regardles of the Result.

Can we just log the error?